db/sql/sql.go

```diff
@@ -705,6 +703,30 @@ func (tx *Transaction) List(s *schema.Schema, filter transaction.Filter, pg *pag
 	return
 }
 
+//List resources in the db
+func (tx *Transaction) List(s *schema.Schema, filter transaction.Filter, pg *pagination.Paginator) (list []*schema.Resource, total uint64, err error) {
+	sql, args, err := buildSelect(s, filter, pg)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return executeSelect(s, filter, sql, args, tx)
+}
+
+//Lock resources in the db
+func (tx *Transaction) LockList(s *schema.Schema, filter transaction.Filter, pg *pagination.Paginator) (list []*schema.Resource, total uint64, err error) {
```

Comment:
I think we shouldn't build with related properties. By default, we shouldn't lock more than specified. Consider optional parameter to pass related resources to lock.